packages/protocol/contracts/L1/libs/LibVerifying.sol

```diff
@@ -166,30 +208,27 @@ library LibVerifying {
         uint256 reward,
         TkoToken tkoToken
     ) private {
-        uint256 start;
+        uint256 offset;
         uint256 count = fc.provers.length;
 
         if (config.enableOracleProver) {
-            start = 1;
+            offset = 1;
             count -= 1;
         }
 
-        uint256 sum = (1 << count) - 1;
-        uint256 weight = 1 << (count - 1);
-        for (uint i = 0; i < count; ++i) {
-            uint256 proverReward = (reward * weight) / sum;
-            if (proverReward == 0) {
-                break;
-            }
+        uint256[] memory weights = getProverRewardBips(config, count);
 
-            if (tkoToken.balanceOf(fc.provers[start + i]) == 0) {
-                // Reduce reward to 1 wei as a penalty if the prover
-                // has 0 TKO balance. This allows the next prover reward
-                // to be fully paid.
-                proverReward = uint256(1);
+        for (uint256 i = 0; i < count; ++i) {
+            uint256 proverReward = (reward * weights[i]) / 10000;
```

Comment:
Not sure if divide `10000` is correct here, since weight is `(a number less than one) * 100 * randomized + (another number less than one) * 100 * (100 - randomized)`. So `weights[i] / 10000` can not represent the actual  % of total rewards that the current prover will get? （Maybe we need to re-calculate the sum of weights here?）

Reply:
I think it is correct, David